rewrite-java/src/main/java/org/openrewrite/java/JavaTemplate.java

```diff
@@ -168,8 +200,8 @@ private void validateImport(String typeName) {
             }
         }
 
-        public Builder javaParser(JavaParser.Builder<?, ?> javaParser) {
-            this.javaParser = javaParser;
+        public Builder javaParser(JavaParser.Builder<?, ?> parser) {
+            this.parser = parser.clone();
```

Comment:
Why is this `clone()` now when it wasn't before?

Reply:
I want to call clone before adding the classpath entry for the `__M__` and `__P__` classes. But it could also be done slightly later. Also, we call clone() later anyway before invoking the parser.